In [1]:
import pandas as pd
import numpy as np

In [8]:
amazon=pd.read_csv('amz_com-ecommerce_sample.csv',encoding= 'unicode_escape')
fipkart=pd.read_csv('flipkart_com-ecommerce_sample.csv',encoding= 'unicode_escape')

In [9]:
amazon.head()

uniq_id            crawl_timestamp  \
0  c2d766ca982eca8304150849735ffef9  2016-03-25 22:59:23 +0000   
1  7f7036a6d550aaa89d34c77bd39a5e48  2016-03-25 22:59:23 +0000   
2  f449ec65dcbc041b6ae5e6a32717d01b  2016-03-25 22:59:23 +0000   
3  0973b37acd0c664e3de26e97e5571454  2016-03-25 22:59:23 +0000   
4  bc940ea42ee6bef5ac7cea3fb5cfbee7  2016-03-25 22:59:23 +0000   

                                         product_url  \
0  http://www.flipkart.com/alisha-solid-women-s-c...   
1  http://www.flipkart.com/fabhomedecor-fabric-do...   
2  http://www.flipkart.com/aw-bellies/p/itmeh4grg...   
3  http://www.flipkart.com/alisha-solid-women-s-c...   
4  http://www.flipkart.com/sicons-all-purpose-arn...   

                            product_name  \
0    Alisha Solid Women's Cycling Shorts   
1    FabHomeDecor Fabric Double Sofa Bed   
2                             AW Bellies   
3    Alisha Solid Women's Cycling Shorts   
4  Sicons All Purpose Arnica Dog Shampoo   

                               product_category_tree               pid  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2FF9KEDEFGF   
1  ["Furniture >> Living Room Furniture >> Sofa B...  SBEEH3QGU7MFYJFY   
2  ["Footwear >> Women's Footwear >> Ballerinas >...  SHOEH4GRSUBJGZXE   
3  ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2F6HUZMQ6SJ   
4  ["Pet Supplies >> Grooming >> Skin & Coat Care...  PSOEH3ZYDMSYARJ5   

   retail_price  discounted_price  \
0           982               438   
1         32143             29121   
2           991               551   
3           694               325   
4           208               258   

                                               image  is_FK_Advantage_product  \
0  ["http://img5a.flixcart.com/image/short/u/4/a/...                    False   
1  ["http://img6a.flixcart.com/image/sofa-bed/j/f...                    False   
2  ["http://img5a.flixcart.com/image/shoe/7/z/z/r...                    False   
3  ["http://img5a.flixcart.com/image/short/6/2/h/...                    False   
4  ["http://img5a.flixcart.com/image/pet-shampoo/...                    False   

                                         description       product_rating  \
0  Key Features of Alisha Solid Women's Cycling S...  No rating available   
1  FabHomeDecor Fabric Double Sofa Bed (Finish Co...  No rating available   
2  Key Features of AW Bellies Sandals Wedges Heel...  No rating available   
3  Key Features of Alisha Solid Women's Cycling S...  No rating available   
4  Specifications of Sicons All Purpose Arnica Do...  No rating available   

        overall_rating         brand  \
0  No rating available        Alisha   
1  No rating available  FabHomeDecor   
2  No rating available            AW   
3  No rating available        Alisha   
4  No rating available        Sicons   

                              product_specifications  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...  
2  {"product_specification"=>[{"key"=>"Ideal For"...  
3  {"product_specification"=>[{"key"=>"Number of ...  
4  {"product_specification"=>[{"key"=>"Pet Type",...

In [13]:
fipkart.head()

uniq_id            crawl_timestamp  \
0  c2d766ca982eca8304150849735ffef9  2016-03-25 22:59:23 +0000   
1  7f7036a6d550aaa89d34c77bd39a5e48  2016-03-25 22:59:23 +0000   
2  f449ec65dcbc041b6ae5e6a32717d01b  2016-03-25 22:59:23 +0000   
3  0973b37acd0c664e3de26e97e5571454  2016-03-25 22:59:23 +0000   
4  bc940ea42ee6bef5ac7cea3fb5cfbee7  2016-03-25 22:59:23 +0000   

                                         product_url  \
0  http://www.flipkart.com/alisha-solid-women-s-c...   
1  http://www.flipkart.com/fabhomedecor-fabric-do...   
2  http://www.flipkart.com/aw-bellies/p/itmeh4grg...   
3  http://www.flipkart.com/alisha-solid-women-s-c...   
4  http://www.flipkart.com/sicons-all-purpose-arn...   

                            product_name  \
0    Alisha Solid Women's Cycling Shorts   
1    FabHomeDecor Fabric Double Sofa Bed   
2                             AW Bellies   
3    Alisha Solid Women's Cycling Shorts   
4  Sicons All Purpose Arnica Dog Shampoo   

                               product_category_tree               pid  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2FF9KEDEFGF   
1  ["Furniture >> Living Room Furniture >> Sofa B...  SBEEH3QGU7MFYJFY   
2  ["Footwear >> Women's Footwear >> Ballerinas >...  SHOEH4GRSUBJGZXE   
3  ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2F6HUZMQ6SJ   
4  ["Pet Supplies >> Grooming >> Skin & Coat Care...  PSOEH3ZYDMSYARJ5   

   retail_price  discounted_price  \
0         999.0             379.0   
1       32157.0           22646.0   
2         999.0             499.0   
3         699.0             267.0   
4         220.0             210.0   

                                               image  is_FK_Advantage_product  \
0  ["http://img5a.flixcart.com/image/short/u/4/a/...                    False   
1  ["http://img6a.flixcart.com/image/sofa-bed/j/f...                    False   
2  ["http://img5a.flixcart.com/image/shoe/7/z/z/r...                    False   
3  ["http://img5a.flixcart.com/image/short/6/2/h/...                    False   
4  ["http://img5a.flixcart.com/image/pet-shampoo/...                    False   

                                         description       product_rating  \
0  Key Features of Alisha Solid Women's Cycling S...  No rating available   
1  FabHomeDecor Fabric Double Sofa Bed (Finish Co...  No rating available   
2  Key Features of AW Bellies Sandals Wedges Heel...  No rating available   
3  Key Features of Alisha Solid Women's Cycling S...  No rating available   
4  Specifications of Sicons All Purpose Arnica Do...  No rating available   

        overall_rating         brand  \
0  No rating available        Alisha   
1  No rating available  FabHomeDecor   
2  No rating available            AW   
3  No rating available        Alisha   
4  No rating available        Sicons   

                              product_specifications  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...  
2  {"product_specification"=>[{"key"=>"Ideal For"...  
3  {"product_specification"=>[{"key"=>"Number of ...  
4  {"product_specification"=>[{"key"=>"Pet Type",...

In [118]:
a=amazon[["product_name","retail_price","discounted_price"]]
f=fipkart[["product_name","retail_price","discounted_price"]]

In [125]:
a

product_name  retail_price  discounted_price
0          alisha solid women's cycling shorts           982               438
1          fabhomedecor fabric double sofa bed         32143             29121
2                                   aw bellies           991               551
3          alisha solid women's cycling shorts           694               325
4        sicons all purpose arnica dog shampoo           208               258
...                                        ...           ...               ...
19995           walldesign small vinyl sticker          1498               876
19996  wallmantra large vinyl stickers sticker          1415              1424
19997  elite collection medium acrylic sticker          1284              1196
19998  elite collection medium acrylic sticker          1492              1364
19999  elite collection medium acrylic sticker          1484              1247

[20000 rows x 3 columns]

In [127]:
f

product_name  retail_price  discounted_price
0          alisha solid women's cycling shorts         999.0             379.0
1          fabhomedecor fabric double sofa bed       32157.0           22646.0
2                                   aw bellies         999.0             499.0
3          alisha solid women's cycling shorts         699.0             267.0
4        sicons all purpose arnica dog shampoo         220.0             210.0
...                                        ...           ...               ...
19995           walldesign small vinyl sticker        1500.0             730.0
19996  wallmantra large vinyl stickers sticker        1429.0            1143.0
19997  elite collection medium acrylic sticker        1299.0             999.0
19998  elite collection medium acrylic sticker        1499.0            1199.0
19999  elite collection medium acrylic sticker        1499.0             999.0

[20000 rows x 3 columns]

In [124]:
a['product_name']=a['product_name'].apply(lambda x:x.lower())
f['product_name']=f['product_name'].apply(lambda x:x.lower())

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\3956782014.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a['product_name']=a['product_name'].apply(lambda x:x.lower())
C:\Users\dell\AppData\Local\Temp\ipykernel_11748\3956782014.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  f['product_name']=f['product_name'].apply(lambda x:x.lower())


In [130]:
a['fipkart_pro_name']=f['product_name']

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\752261380.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a['fipkart_pro_name']=f['product_name']


In [ ]:
a['com'] = np.where(a['product_name'] == a['fipkart_pro_name'], a['product_name'], np.nan)

In [156]:
a[a['com'].isnull()==True]

product_name  retail_price  \
28                            fdt women's leggings pants           698   
16646                 abcd s bean bag  with foam filling          2297   
17944  weavers villa printed set of 3 scarves, stoles...           545   
18005     united bags cross chain 35 l m laptop backpack          1981   
18052                         leaf tork 2.5 l m backpack           981   
...                                                  ...           ...   
19943                         walldesign s vinyl sticker          1288   
19944                 elite collection m acrylic sticker          1282   
19945                 elite collection m acrylic sticker          1488   
19946                         walldesign s vinyl sticker          1281   
19949                         walldesign s vinyl sticker          1290   

       discounted_price                                   fipkart_pro_name  \
28                  362                               fdt women's leggings   
16646              1004             abcd small bean bag  with foam filling   
17944               349  weavers villa printed set of 3 scarves, stoles...   
18005               744  united bags cross chain 35 l medium laptop bac...   
18052              1063                    leaf tork 2.5 l medium backpack   
...                 ...                                                ...   
19943               767                     walldesign small vinyl sticker   
19944              1274            elite collection medium acrylic sticker   
19945              1331            elite collection medium acrylic sticker   
19946               730                     walldesign small vinyl sticker   
19949               741                     walldesign small vinyl sticker   

       com  
28     NaN  
16646  NaN  
17944  NaN  
18005  NaN  
18052  NaN  
...    ...  
19943  NaN  
19944  NaN  
19945  NaN  
19946  NaN  
19949  NaN  

[124 rows x 5 columns]

In [167]:
a['f_retail_price']=f['retail_price']
a['f_discounted_price']=f['discounted_price']

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\1204218144.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a['f_retail_price']=f['retail_price']
C:\Users\dell\AppData\Local\Temp\ipykernel_11748\1204218144.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a['f_discounted_price']=f['discounted_price']


In [203]:
a.isnull().sum()

product_name          0
retail_price          0
discounted_price      0
fipkart_pro_name      0
f_retail_price        0
f_discounted_price    0
dtype: int64

In [184]:
a.dropna(subset=['com'],inplace=True)

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\3040651985.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a.dropna(subset=['com'],inplace=True)


In [189]:
a.reset_index(inplace=True,drop=True)

In [193]:
a.drop(['com'],axis=1, inplace=True)

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\1399659543.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a.drop(['com'],axis=1, inplace=True)


In [194]:
a

product_name  retail_price  \
0          alisha solid women's cycling shorts           982   
1          fabhomedecor fabric double sofa bed         32143   
2                                   aw bellies           991   
3          alisha solid women's cycling shorts           694   
4        sicons all purpose arnica dog shampoo           208   
...                                        ...           ...   
19871           walldesign small vinyl sticker          1498   
19872  wallmantra large vinyl stickers sticker          1415   
19873  elite collection medium acrylic sticker          1284   
19874  elite collection medium acrylic sticker          1492   
19875  elite collection medium acrylic sticker          1484   

       discounted_price                         fipkart_pro_name  \
0                   438      alisha solid women's cycling shorts   
1                 29121      fabhomedecor fabric double sofa bed   
2                   551                               aw bellies   
3                   325      alisha solid women's cycling shorts   
4                   258    sicons all purpose arnica dog shampoo   
...                 ...                                      ...   
19871               876           walldesign small vinyl sticker   
19872              1424  wallmantra large vinyl stickers sticker   
19873              1196  elite collection medium acrylic sticker   
19874              1364  elite collection medium acrylic sticker   
19875              1247  elite collection medium acrylic sticker   

       f_retail_price  f_discounted_price  
0               999.0               379.0  
1             32157.0             22646.0  
2               999.0               499.0  
3               699.0               267.0  
4               220.0               210.0  
...               ...                 ...  
19871          1500.0               730.0  
19872          1429.0              1143.0  
19873          1299.0               999.0  
19874          1499.0              1199.0  
19875          1499.0               999.0  

[19876 rows x 6 columns]

In [198]:
a[a['f_retail_price'].isnull()==True]

product_name  retail_price  \
12               sicons all purpose tea tree dog shampoo           -11   
21                   alisha solid women's cycling shorts            -2   
75                   eurospa cotton terry face towel set           -15   
811                    fundoo t printed men's track suit           -11   
1317   techware microwavable tea cups wf13115 - purpl...            -2   
...                                                  ...           ...   
16760               muchmore alloy copper charm bracelet            -1   
17632                                friction men's vest           -11   
19518                      karishma women's a-line dress           -17   
19574                l'appel du vide women's shift dress           -10   
19597        insense by shoppers stop women's maxi dress           -14   

       discounted_price                                   fipkart_pro_name  \
12                    0            sicons all purpose tea tree dog shampoo   
21                    0                alisha solid women's cycling shorts   
75                    0                eurospa cotton terry face towel set   
811                   0                  fundoo t printed men's track suit   
1317                  0  techware microwavable tea cups wf13115 - purpl...   
...                 ...                                                ...   
16760                 0               muchmore alloy copper charm bracelet   
17632                 0                                friction men's vest   
19518                 0                      karishma women's a-line dress   
19574                 0                l'appel du vide women's shift dress   
19597                 0        insense by shoppers stop women's maxi dress   

       f_retail_price  f_discounted_price  
12                NaN                 NaN  
21                NaN                 NaN  
75                NaN                 NaN  
811               NaN                 NaN  
1317              NaN                 NaN  
...               ...                 ...  
16760             NaN                 NaN  
17632             NaN                 NaN  
19518             NaN                 NaN  
19574             NaN                 NaN  
19597             NaN                 NaN  

[78 rows x 6 columns]

In [199]:
a.dropna(subset=['f_retail_price'],inplace=True)

C:\Users\dell\AppData\Local\Temp\ipykernel_11748\3305635040.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  a.dropna(subset=['f_retail_price'],inplace=True)


In [202]:
a

product_name  retail_price  \
0          alisha solid women's cycling shorts           982   
1          fabhomedecor fabric double sofa bed         32143   
2                                   aw bellies           991   
3          alisha solid women's cycling shorts           694   
4        sicons all purpose arnica dog shampoo           208   
...                                        ...           ...   
19793           walldesign small vinyl sticker          1498   
19794  wallmantra large vinyl stickers sticker          1415   
19795  elite collection medium acrylic sticker          1284   
19796  elite collection medium acrylic sticker          1492   
19797  elite collection medium acrylic sticker          1484   

       discounted_price                         fipkart_pro_name  \
0                   438      alisha solid women's cycling shorts   
1                 29121      fabhomedecor fabric double sofa bed   
2                   551                               aw bellies   
3                   325      alisha solid women's cycling shorts   
4                   258    sicons all purpose arnica dog shampoo   
...                 ...                                      ...   
19793               876           walldesign small vinyl sticker   
19794              1424  wallmantra large vinyl stickers sticker   
19795              1196  elite collection medium acrylic sticker   
19796              1364  elite collection medium acrylic sticker   
19797              1247  elite collection medium acrylic sticker   

       f_retail_price  f_discounted_price  
0               999.0               379.0  
1             32157.0             22646.0  
2               999.0               499.0  
3               699.0               267.0  
4               220.0               210.0  
...               ...                 ...  
19793          1500.0               730.0  
19794          1429.0              1143.0  
19795          1299.0               999.0  
19796          1499.0              1199.0  
19797          1499.0               999.0  

[19798 rows x 6 columns]

In [208]:
a.to_csv('Task_2.csv')

In [209]:
import pickle
pickle.dump(a.to_dict(),open('prices.pkl','wb'))

# there are 124 nan values in flipkart retail price and discounted price and in amazon the same product have negative values so
# we drop all nan values values 
# there are more than 7000 dupicated values in both amazon and flipkart product name may be they are of diiferrent colors
# there are 124 unmatched products from flipkart and amazon
# there are like small is wriiten as 's' in flipkart but small in amazon but still we remove this like items also
# we use numpy.where to find matched product name from both side